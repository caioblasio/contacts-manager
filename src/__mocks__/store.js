const contactsState = require("providers/Contacts/reducer").initialState;

const history = require("history").createBrowserHistory();
const configureStore = require("store").default;

const rootState = {
  contacts: contactsState
};

const store = configureStore(rootState);

module.exports = store;
