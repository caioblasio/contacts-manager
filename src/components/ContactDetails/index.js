import React from "react";
import PropTypes from "prop-types";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Avatar from "@material-ui/core/Avatar";
import Divider from "@material-ui/core/Divider";
import AccountCircleIcon from "@material-ui/icons/AccountCircle";
import EmailIcon from "@material-ui/icons/Email";
import PhoneIcon from "@material-ui/icons/Phone";
import HttpIcon from "@material-ui/icons/Http";
import useStyles from "./styles";

const ContactDetails = ({ contact }) => {
  const classes = useStyles();
  return (
    <Card className={classes.root}>
      <CardContent>
        <div className={classes.head}>
          <Avatar className={classes.avatar}>
            {contact.name.split("")[0].toUpperCase()}
          </Avatar>
          <Typography className={classes.title} variant="h4" component="h3">
            {contact.name}
          </Typography>
        </div>
        <Divider />
        <div className={classes.content}>
          <Typography color="textSecondary">ID: {contact.id}</Typography>
          <div className={classes.info}>
            <AccountCircleIcon />
            <Typography variant="body2" component="p">
              {contact.username}
            </Typography>
          </div>
          <div className={classes.info}>
            <EmailIcon />
            <Typography variant="body2" component="p">
              {contact.email}
            </Typography>
          </div>
          <div className={classes.info}>
            <PhoneIcon />
            <Typography variant="body2" component="p">
              <a href={`tel:${contact.phone}`}>{contact.phone}</a>
            </Typography>
          </div>
          <div className={classes.info}>
            <HttpIcon />
            <Typography variant="body2" component="p">
              <a href={`//${contact.website}`} target="_blank">
                {contact.website}
              </a>
            </Typography>
          </div>
        </div>
      </CardContent>
    </Card>
  );
};

ContactDetails.propTypes = {
  contact: PropTypes.object.isRequired
};

export default ContactDetails;
