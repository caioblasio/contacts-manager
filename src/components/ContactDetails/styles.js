import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  root: {},
  head: {
    display: "flex",
    alignItems: "center",
    paddingBottom: theme.spacing(2)
  },
  avatar: {
    width: theme.spacing(8),
    height: theme.spacing(8),
    marginRight: theme.spacing(4)
  },
  content: {
    paddingTop: theme.spacing(2)
  },
  info: {
    display: "flex",
    marginTop: theme.spacing(2),
    alignItems: "center",
    "& :first-child": {
      marginRight: theme.spacing(2)
    }
  }
}));

export default useStyles;
