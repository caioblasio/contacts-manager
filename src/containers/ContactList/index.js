import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { getData } from "providers/Contacts/selectors";
import ContactList from "components/ContactList";

export const Container = props => {
  return <ContactList {...props} />;
};

const mapStateToProps = createStructuredSelector({
  data: getData()
});

Container.propTypes = {
  data: PropTypes.array
};

export default connect(mapStateToProps)(Container);
