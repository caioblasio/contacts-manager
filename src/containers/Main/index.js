import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Route, Switch } from "react-router-dom";
import CssBaseline from "@material-ui/core/CssBaseline";
import Home from "pages/Home";
import Details from "pages/Details";
import New from "pages/New";
import Topbar from "components/Topbar";
import { fetchContacts } from "providers/Contacts/actions";

export const Container = ({ fetchContacts }) => {
  useEffect(() => {
    fetchContacts();
  });
  return (
    <main className="main">
      <CssBaseline />
      <Topbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route
          exact
          path="/contact/:id"
          render={props => <Details contactId={props.match.params.id} />}
        />
        <Route exact path="/new" component={New} />
      </Switch>
    </main>
  );
};

function mapDispatchToProps(dispatch) {
  return {
    fetchContacts: () => dispatch(fetchContacts())
  };
}

Container.propTypes = {
  fetchContacts: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(Container);
