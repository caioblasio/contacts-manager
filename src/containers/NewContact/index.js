import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import NewContact from "components/NewContact";
import { addContact } from "providers/Contacts/actions";
import { push } from "providers/Router/actions";

export const Container = props => {
  return <NewContact {...props} />;
};

const mapDispatchToProps = dispatch => ({
  addContact: contact => dispatch(addContact(contact)),
  routePush: url => dispatch(push(url))
});

Container.propTypes = {
  addContact: PropTypes.func.isRequired,
  routePush: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(Container);
