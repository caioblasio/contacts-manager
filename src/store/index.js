import { createStore, applyMiddleware, compose } from "redux";
import { createBrowserHistory } from "history";
import { routerMiddleware } from "connected-react-router/immutable";
import createSagaMiddleware from "redux-saga";
import createRootReducer from "./reducers";
import rootSaga from "./sagas";

export const history = createBrowserHistory();

export default function configureStore(initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};
  if (ENVIRONMENT !== "production" && typeof window === "object") {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        name: "Redux",
        realtime: true,
        trace: true,
        traceLimit: 25
      });
    }
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const middlewares = [sagaMiddleware, routerMiddleware(history)];
  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createRootReducer(history),
    initialState,
    composeEnhancers(...enhancers)
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
