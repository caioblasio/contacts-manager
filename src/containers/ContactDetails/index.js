import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { getContact } from "providers/Contacts/selectors";
import ContactDetails from "components/ContactDetails";

export const Container = ({ contact }) => {
  return contact ? (
    <ContactDetails contact={contact} />
  ) : (
    "Contact not found! :("
  );
};

const mapStateToProps = (_state, { id }) => {
  return createStructuredSelector({
    contact: getContact(id),
  });
};

Container.propTypes = {
  contact: PropTypes.object,
};

export default connect(mapStateToProps)(Container);
