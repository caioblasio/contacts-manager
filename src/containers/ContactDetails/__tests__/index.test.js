import React from "react";
import { Provider } from "react-redux";
import { render, queryByText } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Container as ContactDetailsContainer } from "containers/ContactDetails";

import store from "__mocks__/store";
import { data } from "__mocks__/api";

describe("Given <ContactDetailsContainer />", () => {
  let defaultProps;
  let component;
  let Container;

  defaultProps = {
    contact: data[0]
  };

  beforeAll(() => {
    Container = props => (
      <Provider store={store}>
        <MemoryRouter>
          <ContactDetailsContainer {...defaultProps} {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  describe("When it initializes", () => {
    beforeEach(() => {
      const { container } = render(<Container />);
      component = container;
    });

    it("Then should match the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
