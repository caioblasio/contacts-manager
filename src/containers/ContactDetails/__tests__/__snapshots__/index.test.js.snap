// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ContactDetailsContainer /> When it initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <div
        class="makeStyles-head-2"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-3 MuiAvatar-colorDefault"
        >
          L
        </div>
        <h3
          class="MuiTypography-root MuiTypography-h4"
        >
          Leanne Graham
        </h3>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="makeStyles-content-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: 
          1
        </p>
        <div
          class="makeStyles-info-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Bret
          </p>
        </div>
        <div
          class="makeStyles-info-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Sincere@april.biz
          </p>
        </div>
        <div
          class="makeStyles-info-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            <a
              href="tel:1-770-736-8031 x56442"
            >
              1-770-736-8031 x56442
            </a>
          </p>
        </div>
        <div
          class="makeStyles-info-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.5 11h-2V9H1v6h1.5v-2.5h2V15H6V9H4.5v2zm2.5-.5h1.5V15H10v-4.5h1.5V9H7v1.5zm5.5 0H14V15h1.5v-4.5H17V9h-4.5v1.5zm9-1.5H18v6h1.5v-2h2c.8 0 1.5-.7 1.5-1.5v-1c0-.8-.7-1.5-1.5-1.5zm0 2.5h-2v-1h2v1z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            <a
              href="//hildegard.org"
              target="_blank"
            >
              hildegard.org
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
